### PR TITLE
[SPARK-7243][SQL] Reduce  size for Contingency Tables in DataFrames

### PR DESCRIPTION
**Pull request by brkyvz**
Reduced take size from 1e8 to 1e6.

cc @rxin

### CODE REVIEW

**Comment by brkyvz** [comment]
retest this please

File: sql/core/src/main/scala/org/apache/spark/sql/execution/stat/StatFunctions.scala [hunk 1]
```diff
@@ -102,9 +102,9 @@ private[sql] object StatFunctions extends Logging {
   /** Generate a table of frequencies for the elements of two columns. */
   private[sql] def crossTabulate(df: DataFrame, col1: String, col2: String): DataFrame = {
     val tableName = s"${col1}_$col2"
-    val counts = df.groupBy(col1, col2).agg(col(col1), col(col2), count("*")).take(1e8.toInt)
-    if (counts.length == 1e8.toInt) {
-      logWarning("The maximum limit of 1e8 pairs have been collected, which may not be all of " +
+    val counts = df.groupBy(col1, col2).agg(col(col1), col(col2), count("*")).take(1e6.toInt)
+    if (counts.length == 1e6.toInt) {
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/stat/StatFunctions.scala]: we should also update the user facing javadoc/python docstring to say we get max of 1 million entries ...

**Comment by brkyvz** [comment]
Retest this please

**Comment by brkyvz** [comment]
retest this please

**Comment by rxin** [comment]
Merging. Thanks.

**Pull request closed without merging**